### PR TITLE
Add a dedicated TransportRemoteInfoAction for consistency

### PR DESCRIPTION
**Pull request by s1monw**
All our actions that are invoked from rest actions have corresponding
transport actions. This adds the transport action for RestRemoteClusterInfoAction
for consistency.

Relates to #23969

### CODE REVIEW

File: core/src/main/java/org/elasticsearch/action/admin/cluster/remote/RemoteInfoRequestBuilder.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.action.admin.cluster.remote;
+
+import org.elasticsearch.action.ActionRequestBuilder;
+import org.elasticsearch.client.ElasticsearchClient;
+
+public class RemoteInfoRequestBuilder extends ActionRequestBuilder<RemoteInfoRequest, RemoteInfoResponse, RemoteInfoRequestBuilder> {
```
Comment (jaymode) [inline core/src/main/java/org/elasticsearch/action/admin/cluster/remote/RemoteInfoRequestBuilder.java]: make this class final as well?

File: core/src/main/java/org/elasticsearch/action/admin/cluster/remote/RemoteInfoResponse.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        super.writeTo(out);
+        out.writeList(infos);
+    }
+
+    @Override
+    public void readFrom(StreamInput in) throws IOException {
+        super.readFrom(in);
```
Comment (jaymode) [inline core/src/main/java/org/elasticsearch/action/admin/cluster/remote/RemoteInfoResponse.java]: missing ```infos = in.readList();``` ?

File: core/src/main/java/org/elasticsearch/action/admin/cluster/remote/TransportRemoteInfoAction.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+                                     SearchTransportService searchTransportService) {
+        super(settings, RemoteInfoAction.NAME, threadPool, transportService, actionFilters, indexNameExpressionResolver,
+            RemoteInfoRequest::new);
+        this.remoteClusterService = searchTransportService.getRemoteClusterService();
+    }
+
+    @Override
+    protected void doExecute(RemoteInfoRequest remoteInfoRequest, ActionListener<RemoteInfoResponse> listener) {
+        remoteClusterService.getRemoteConnectionInfos(ActionListener.wrap(remoteConnectionInfos
+                -> listener.onResponse(new RemoteInfoResponse(new ArrayList<>(remoteConnectionInfos))),
```
Comment (jaymode) [inline core/src/main/java/org/elasticsearch/action/admin/cluster/remote/TransportRemoteInfoAction.java]: maybe make wrap the list in an unmodifiable list?

**Review by jaymode — CHANGES_REQUESTED** [review]

File: core/src/main/java/org/elasticsearch/action/admin/cluster/remote/RemoteInfoResponse.java (hunk 2 above)
Comment (s1monw) [inline core/src/main/java/org/elasticsearch/action/admin/cluster/remote/RemoteInfoResponse.java]: phew yeah 💃

**Review by jaymode — APPROVED** [review]
LGTM

**Pull request merged**